### ISSUE TITLE
Support memory profiling with jemalloc_pprof

### ISSUE THREAD

**Issue by joshg-ec** [grafana/pyroscope-rs#142]
Rust heap profile support was recently implemented in the [`jemalloc_pprof` crate](https://crates.io/crates/jemalloc_pprof). Add support for uploading heap profiles to Pyroscope.

**Comment by GerryLon** [comment]
I second the motion

**Comment by korniltsev** [comment]
I took a look at the jemalloc_pprof crate
It produces pprof without symbols.
I've created a quick dirty POC of patched jemalloc_pprof to symbolize pprofs on the fly.
https://github.com/korniltsev/rust-jemalloc-pprof/pull/1
And used it in a tiny app for demo purposes
https://github.com/korniltsev/broot/pull/1The results are great and terrible at the same time;
Great - it seem to work.
Terrible - it consumes a lot of memory, like 99 % of the memory is from symbolizing, maybe I've programmed it wrong - did not try to investigate.I also did not demangle rust symbols, it will also take some resources and make it even more inefficient.This may be a good reason to support ingesting symbols separately. https://github.com/grafana/pyroscope/pull/3799 

tldr: it's not suitable for use as continuous profiling with the current version of pyroscope and requires some efforts either on backend symbolization or efficient client symbolization.
